server/service.go

```diff
@@ -601,10 +630,23 @@ func (m *BoostService) processCapellaPayload(w http.ResponseWriter, req *http.Re
 		relays = m.relays
 	}
 
+	// Get the uid for this slot
+	uid := ""
+	m.slotUIDLock.Lock()
+	if m.slotUID.slot == uint64(payload.Message.Slot) {
+		uid = m.slotUID.uid.String()
```

Comment:
this works but I will note is somewhat fragile -- rather than keeping a single UUID at a time, the node could just keep a map keyed by the `getHeader` request data `(slot, pubkey, parentHash)` along w/ the UUID

there could be a background goroutine to clean up this in-memory data structure so it doesn't just grow unbounded over time

I am thinking it would be good to have more precise keying in the situation where this feature would be most important -- debugging relay behavior in an elevated forking or reorging regime

Reply:
yeah, i guess you are right that this might be even more useful!

Reply:
i totally get where @ralexstokes is coming from here, but its not clear to me that adding the complexity will really make debugging much easier. we log the UUID on the validator side and we will also pick it up on the relay side and that should be plenty to get the debugging done. the slot and pubkey will be in the relay logs right by the UUID. also if there is an attacker calling getHeader with different parentHashes, then (1) we catch that on the relay end and log at the WARN level, and (2) they probably will modify their boost client to not send the UUID at all!

Reply:
unless there's further comments, we'll merge the current implementation and extend the granularity once needed